### PR TITLE
Store benchmark data

### PR DESCRIPTION
**Pull request by dnadales**
Turn the benchmark data into a JSON file that can be fed to https://github.com/benchmark-action/github-action-benchmark

Reduce the number of mempool benchmarks we run to 2.

# Checklist

- Branch
    - [ ] Commit sequence broadly makes sense
    - [ ] Commits have useful messages
    - [ ] The documentation has been properly updated
    - [ ] New tests are added if needed and existing tests are updated
    - [ ] Any changes affecting Consensus packages must have an entry in the appropriate `changelog.d` directory created using [`scriv`](https://github.com/input-output-hk/scriv). If in doubt, see the [Consensus release process](../ouroboros-consensus/docs/ReleaseProcess.md).
    - [ ] Any changes in the Consensus API (every exposed function, type or module) that has changed its name, has been deleted, has been moved, or altered in some other significant way must leave behind a `DEPRECATED` warning that notifies downstream consumers.
    - [ ] If this branch changes Network and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../docs/interface-CHANGELOG.md)
    - [ ] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [ ] Reviewer requested

### CODE REVIEW

File: ouroboros-consensus-test/bench-mempool/Main.hs [hunk 1]
```diff
@@ -18,37 +33,116 @@ import           Bench.Consensus.MempoolWithMockedLedgerItf
...
+                  , testCase "txs length"    $ length txs @?= n
+                  ]
+                )
+              where
+                testAddTxs mempool txs = do
+                    run mempool txs
+                    mempoolTxs <- getTxs mempool
+                    mempoolTxs @?= txsAddedInCmds txs
+
+    parseBenchmarkResults csvFilePath = do
```
Comment (abailly-iohk) [inline ouroboros-consensus-test/bench-mempool/Main.hs:74]: cassava is already in the dependencies of ouroboros-network so it won't add much and will be more robust than hand-made parsing.

Comment (abailly-iohk) [reply ouroboros-consensus-test/bench-mempool/Main.hs:74]: I think if you do 
 
```
data BenchPoint = ...
    deriving (Generic, ToJSON, FromRow)
```

converting from CSV to JSON can boil down to a couple lines

**Review by abailly-iohk — APPROVED** [review]
LGTM. I would use proper CSV parsing in a later PR

File: ouroboros-consensus-test/bench-mempool/Main.hs (hunk 1 above)
Comment (dnadales) [reply ouroboros-consensus-test/bench-mempool/Main.hs:74]: We need to do some picoseconds conversions and processing of the benchmark names, so I'd rather avoid this generic instance.

Comment (dnadales) [reply ouroboros-consensus-test/bench-mempool/Main.hs:74]: > cassava is already in the dependencies of ouroboros-network so it won't add much and will be more robust than hand-made parsing.

Oh, good point. Let me try that.

Comment (dnadales) [reply ouroboros-consensus-test/bench-mempool/Main.hs:74]: Done https://github.com/input-output-hk/ouroboros-network/pull/4422/commits/12962af5d3e7fab87e1565969aa061182d68526e

File: ouroboros-consensus-test/bench-mempool/Main.hs [hunk 2]
```diff
@@ -18,37 +33,95 @@ import           Bench.Consensus.MempoolWithMockedLedgerItf
...
+            (adjustedMean, unit) = first round
+                                 $ convertPicosecondsWithUnit
+                                 $ fromInteger
+                                 $ textToInt mean
+              where
+                textToInt = either error fst . Text.Read.decimal
+
+            -- Convert a number of picoseconds to the largest time unit that
+            -- makes the conversion greater or equal than one.
+            convertPicosecondsWithUnit :: Double -> (Double, String)
```
Comment (abailly-iohk) [inline ouroboros-consensus-test/bench-mempool/Main.hs:105]: Out of curiosity, why is this conversion needed? To scale the benchmarks to the best unit?

**Review by abailly-iohk — APPROVED** [review]

File: ouroboros-consensus-test/bench-mempool/Main.hs (hunk 2 above)
Comment (dnadales) [reply ouroboros-consensus-test/bench-mempool/Main.hs:105]: Right. This is what `Tasty.Benchmark` seems to be doing as well.

**Comment by dnadales** [comment]
bors retry

**Pull request merged**